Add GHA CI workflow

arueckauer opened the pull request:
|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

Replace TravisCI build with GitHub Action CI workflow.

arueckauer committed:
Add GHA CI workflow

Signed-off-by: Andi Rückauer <arueckauer@gmail.com>

arueckauer committed:
Recreate composer.lock under PHP 7.3

Signed-off-by: Andi Rückauer <arueckauer@gmail.com>

arueckauer committed:
Replace ocramius/package-versions with composer/package-versions-deprecated due to compatibility issues

Signed-off-by: Andi Rückauer <arueckauer@gmail.com>

froschdesign commented on README.md:
```diff
@@ -1,7 +1,6 @@
 
-[![Build Status](https://travis-ci.com/mezzio/mezzio-tooling.svg?branch=master)](https://travis-ci.com/mezzio/mezzio-tooling)
-[![Coverage Status](https://coveralls.io/repos/github/mezzio/mezzio-tooling/badge.svg?branch=master)](https://coveralls.io/github/mezzio/mezzio-tooling?branch=master)
+[![Build Status](https://github.com/mezzio/mezzio-tooling/actions/workflows/continous-integration.yml/badge.svg)](https://github.com/mezzio/mezzio-tooling/actions/workflows/continous-integration.yml)
```
```suggestion
[![Build Status](https://github.com/mezzio/mezzio-tooling/actions/workflows/continuous-integration.yml/badge.svg)](https://github.com/mezzio/mezzio-tooling/actions/workflows/continuous-integration.yml)
```
You can check the preview via "View file" if the image file works: https://github.com/mezzio/mezzio-tooling/blob/007d8f8a7f976a87511759205c306a478f2ba258/README.md

arueckauer replied:
Good catch!

arueckauer committed:
Fix typo

Signed-off-by: Andi Rückauer <arueckauer@gmail.com>

froschdesign commented on composer.json:
```diff
@@ -27,6 +27,7 @@
     "require": {
         "php": "^7.1",
```
You can use PHP version 7.3 here because of #7

arueckauer replied:
Done.

arueckauer committed:
Fix build by dropping support for PHP prior 7.3 as of #7

Signed-off-by: Andi Rückauer <arueckauer@gmail.com>

weierophinney committed:
qa: revert bump in PHP constraint

- Reverts bump of PHP constraint back to `^7.1`
- Skip locked runs on PHP 7.1 and 7.2

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>
